### ISSUE TITLE
highgui qt compile error multiple definition of

### ISSUE THREAD

**Issue by arpu** [opencv/opencv#9813]
opencv master 29b0ef44aa24e237498b032c2ab5adc63ffafadb  
g++ (GCC) 7.2.1 20170915 (Red Hat 7.2.1-2)
Using Qt version 5.7.1 in /usr/lib64

`CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11CvButtonbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvButtonbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11CvButtonbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::metaObject() const':
mocs_compilation.cpp:(.text._ZNK11GuiReceiver10metaObjectEv+0x0): multiple definition of `GuiReceiver::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK11GuiReceiver10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x200): multiple definition of `GuiReceiver::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x200): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::metaObject() const':
mocs_compilation.cpp:(.text._ZNK11CvButtonbar10metaObjectEv+0x0): multiple definition of `CvButtonbar::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK11CvButtonbar10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x1c0): multiple definition of `CvButtonbar::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x1c0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::metaObject() const':
mocs_compilation.cpp:(.text._ZNK12CvPushButton10metaObjectEv+0x0): multiple definition of `CvPushButton::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK12CvPushButton10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x180): multiple definition of `CvPushButton::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x180): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::metaObject() const':
mocs_compilation.cpp:(.text._ZNK10CvCheckBox10metaObjectEv+0x0): multiple definition of `CvCheckBox::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK10CvCheckBox10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x140): multiple definition of `CvCheckBox::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x140): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::metaObject() const':
mocs_compilation.cpp:(.text._ZNK13CvRadioButton10metaObjectEv+0x0): multiple definition of `CvRadioButton::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK13CvRadioButton10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x100): multiple definition of `CvRadioButton::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x100): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::metaObject() const':
mocs_compilation.cpp:(.text._ZNK10CvTrackbar10metaObjectEv+0x0): multiple definition of `CvTrackbar::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK10CvTrackbar10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0xc0): multiple definition of `CvTrackbar::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0xc0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::metaObject() const':
mocs_compilation.cpp:(.text._ZNK15CvWinProperties10metaObjectEv+0x0): multiple definition of `CvWinProperties::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK15CvWinProperties10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x80): multiple definition of `CvWinProperties::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x80): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::metaObject() const':
mocs_compilation.cpp:(.text._ZNK8CvWindow10metaObjectEv+0x0): multiple definition of `CvWindow::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK8CvWindow10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x40): multiple definition of `CvWindow::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x40): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::metaObject() const':
mocs_compilation.cpp:(.text._ZNK15DefaultViewPort10metaObjectEv+0x0): multiple definition of `DefaultViewPort::metaObject() const'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZNK15DefaultViewPort10metaObjectEv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o:(.data.rel.ro+0x0): multiple definition of `DefaultViewPort::staticMetaObject'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:(.data.rel.ro+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN11CvButtonbar11qt_metacastEPKc+0x0): multiple definition of `CvButtonbar::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11CvButtonbar11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN10CvTrackbar11qt_metacastEPKc+0x0): multiple definition of `CvTrackbar::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvTrackbar11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvButtonbar::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11CvButtonbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvButtonbar::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11CvButtonbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvTrackbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvTrackbar::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvTrackbar18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvTrackbar::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvTrackbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvTrackbar::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvTrackbar11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN8CvWindow18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWindow::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN8CvWindow18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15DefaultViewPort18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `DefaultViewPort::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15DefaultViewPort18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15CvWinProperties18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWinProperties::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15CvWinProperties18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN12CvPushButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvPushButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN12CvPushButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN13CvRadioButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvRadioButton::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN13CvRadioButton18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvCheckBox18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvCheckBox::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvCheckBox18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11GuiReceiver18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): multiple definition of `GuiReceiver::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11GuiReceiver18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN11GuiReceiver11qt_metacastEPKc+0x0): multiple definition of `GuiReceiver::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11GuiReceiver11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `GuiReceiver::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN11GuiReceiver11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `GuiReceiver::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN11GuiReceiver11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN12CvPushButton11qt_metacastEPKc+0x0): multiple definition of `CvPushButton::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN12CvPushButton11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvPushButton::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN12CvPushButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvPushButton::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN12CvPushButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN10CvCheckBox11qt_metacastEPKc+0x0): multiple definition of `CvCheckBox::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvCheckBox11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvCheckBox::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN10CvCheckBox11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvCheckBox::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN10CvCheckBox11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN13CvRadioButton11qt_metacastEPKc+0x0): multiple definition of `CvRadioButton::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN13CvRadioButton11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvRadioButton::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN13CvRadioButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvRadioButton::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN13CvRadioButton11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN15CvWinProperties11qt_metacastEPKc+0x0): multiple definition of `CvWinProperties::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15CvWinProperties11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN8CvWindow11qt_metacastEPKc+0x0): multiple definition of `CvWindow::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN8CvWindow11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWinProperties::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15CvWinProperties11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWinProperties::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15CvWinProperties11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `CvWindow::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN8CvWindow11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `CvWindow::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN8CvWindow11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::qt_metacast(char const*)':
mocs_compilation.cpp:(.text._ZN15DefaultViewPort11qt_metacastEPKc+0x0): multiple definition of `DefaultViewPort::qt_metacast(char const*)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15DefaultViewPort11qt_metacastEPKc+0x0): first defined here
CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: In function `DefaultViewPort::qt_metacall(QMetaObject::Call, int, void**)':
mocs_compilation.cpp:(.text._ZN15DefaultViewPort11qt_metacallEN11QMetaObject4CallEiPPv+0x0): multiple definition of `DefaultViewPort::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o:moc_window_QT.cpp:(.text._ZN15DefaultViewPort11qt_metacallEN11QMetaObject4CallEiPPv+0x0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [modules/highgui/CMakeFiles/opencv_highgui.dir/build.make:257: lib/libopencv_highgui.so.3.3.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:5408: modules/highgui/CMakeFiles/opencv_highgui.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
`

**Comment by alalek** [comment]
I can't reproduce problem.
1. Try to build from scratch (clean build directory - CMake cache some things - we need reproducible results)
2. Add your build configuration from CMake output (stdout).

BTW, contents of mocs_compilation.cpp file:
```
$ cat modules/highgui/opencv_highgui_autogen/mocs_compilation.cpp 
/* This file is autogenerated, do not edit*/
#include "UVLADIE3JM/moc_window_QT.cpp"
```
There is no reason to build both .cpp files

**Comment by arpu** [comment]
same problem on clean install
complete build output http://lpaste.net/6966961691741388800

`cat modules/highgui/opencv_highgui_autogen/mocs_compilation.cpp 
/* This file is autogenerated, do not edit*/
#include "UVLADIE3JM/moc_window_QT.cpp"
`

**Comment by alalek** [comment]
Thank you for update!
Could you try to add this CMake option: `-DENABLE_PRECOMPILED_HEADERS=OFF`? Looks like QT tries to wrap "pch" targets too.

**Comment by arpu** [comment]
http://lpaste.net/5444999984128720896
complete build log, same problem

**Comment by alalek** [comment]
Build log blames on file `modules/highgui/CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o`
But I have this (no such file):
```
$ ll modules/highgui/CMakeFiles/opencv_highgui.dir/src/
total 376
-rw-rw-r--. 1 alalek alalek  21464 Oct  9 21:51 roiSelector.cpp.o
-rw-rw-r--. 1 alalek alalek  44808 Oct  9 21:51 window.cpp.o
-rw-rw-r--. 1 alalek alalek 315248 Oct  9 21:51 window_QT.cpp.o
```
and this in opencv sources:
```
$ ll ../opencv/modules/highgui/src/
total 372
drwxrwxr-x. 1 alalek alalek    56 Apr 13 18:14 files_Qt
-rw-rw-r--. 1 alalek alalek  4600 Sep 25 15:15 precomp.hpp
-rw-rw-r--. 1 alalek alalek  6962 Aug 28 19:26 roiSelector.cpp
-rw-rw-r--. 1 alalek alalek 32497 Sep 25 15:15 window_carbon.cpp
-rw-rw-r--. 1 alalek alalek 32923 Aug 28 19:26 window_cocoa.mm
-rw-rw-r--. 1 alalek alalek 19351 Oct  5 14:14 window.cpp
-rw-rw-r--. 1 alalek alalek 56629 Oct  5 14:14 window_gtk.cpp
-rw-rw-r--. 1 alalek alalek 87587 Oct  5 14:14 window_QT.cpp
-rw-rw-r--. 1 alalek alalek 16115 Aug 18 18:09 window_QT.h
-rw-rw-r--. 1 alalek alalek   800 Apr 13 18:14 window_QT.qrc
-rw-rw-r--. 1 alalek alalek 71244 Oct  5 14:14 window_w32.cpp
-rw-rw-r--. 1 alalek alalek 11984 Sep 25 15:15 window_winrt_bridge.cpp
-rw-rw-r--. 1 alalek alalek  8009 Aug  2 16:28 window_winrt_bridge.hpp
-rw-rw-r--. 1 alalek alalek  8467 Aug  2 16:28 window_winrt.cpp
```

Please grab command line (and CMake targets) which are used to build this file:
```
... build error ...
$ rm modules/highgui/CMakeFiles/opencv_highgui.dir/src/moc_window_QT.cpp.o
$ make VERBOSE=1
```

Also makefile rules may help:
```
$ grep -R moc_window_QT.cpp ./modules/highgui/
./modules/highgui/CMakeFiles/opencv_highgui.dir/CXX.includecache:/home/alalek/projects/opencv/build/opencv/modules/highgui/opencv_highgui_autogen/UVLADIE3JM/moc_window_QT.cpp
./modules/highgui/CMakeFiles/opencv_highgui.dir/CXX.includecache:UVLADIE3JM/moc_window_QT.cpp
./modules/highgui/CMakeFiles/opencv_highgui.dir/CXX.includecache:/home/alalek/projects/opencv/build/opencv/modules/highgui/opencv_highgui_autogen/UVLADIE3JM/moc_window_QT.cpp
./modules/highgui/CMakeFiles/opencv_highgui.dir/depend.internal: /home/alalek/projects/opencv/build/opencv/modules/highgui/opencv_highgui_autogen/UVLADIE3JM/moc_window_QT.cpp
./modules/highgui/CMakeFiles/opencv_highgui.dir/depend.make:modules/highgui/CMakeFiles/opencv_highgui.dir/opencv_highgui_autogen/mocs_compilation.cpp.o: modules/highgui/opencv_highgui_autogen/UVLADIE3JM/moc_window_QT.cpp
./modules/highgui/opencv_highgui_autogen/mocs_compilation.cpp:#include "UVLADIE3JM/moc_window_QT.cpp"
```

**Comment by alalek** [comment]
Well, I was on wrong OpenCV commit. Problem is reproduced.

**Comment by alalek** [comment]
@arpu Could you try this fix: #9815 ?

**Comment by arpu** [comment]
yes this fixed the build problem. thx a lot to look at this 
bug report should be closed
